src/compiler/scala/tools/nsc/typechecker/Infer.scala

```diff
@@ -574,6 +574,13 @@ trait Infer extends Checkable {
           }
         )
       }
+      if (settings.warnInferAny) {
+        foreachWithIndex(targs){ (targ, idx) =>
+          val tparam = tparams(idx)
+          if (!tparam.isMonomorphicType && !targ.isHigherKinded)
+            reporter.warning(argumentPosition(idx), s"type inferrence relied on kind-polymorphic nature of inferred type `$targ` to conform to type parameter ${tparam.defString}")
```

Comment:
The typo stemmed from a copy/paste/edit from the neighbouring warning. Just pushed a a Webster's certified version.